src/yarp.c

```diff
@@ -734,10 +739,9 @@ lex_token_type(yp_parser_t *parser) {
             return YP_TOKEN_TILDE_AT;
           return YP_TOKEN_TILDE;
 
-        // TODO
         case '\\':
-          return YP_TOKEN_INVALID;
-
+          advance_blank_characters(parser);
+          break;
```

Comment:
We need to return something here and not just `break`. I think if we `break` then it's going to return `YP_TOKEN_INVALID` by default. If we're truly just skipping past these characters, maybe we should just return `lex_token_type(parser)`?

Reply:
I changed it and it's now a little closer to being correct. However, on parsing tests are failing with
```
<[[2, 1], :on_tstring_beg, "'"]> expected but was
<[[1, 7], :on_nl, "\n"]>
```
I guess we're not matching Ripper exactly.

Reply:
We probably need to emit a newline token after the `\` is found if there is one. Then in the parser you should `accept(parser, YP_TOKEN_NEWLINE)`